hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -323,21 +315,67 @@ public synchronized void start() {
     }
 
     public void tryConnectToAllClusterMembers(boolean sync) {
-        if (!isSmartRoutingEnabled) {
-            return;
+        if (isSmartRoutingEnabled && sync) {
+            connectToAllClusterMembers();
         }
+    }
 
-        if (sync) {
-            for (Member member : client.getClientClusterService().getMemberList()) {
-                try {
-                    getOrConnectToMember(member, false);
-                } catch (Exception e) {
-                    EmptyStatement.ignore(e);
+    private void clusterThreadLoop() {
```

Comment:
This is started before we call `tryConnectToAllClusterMembers` in the main thread. Just to double check, can there be some racy scenarios?

Reply:
It looks like there can be. We need to make sure that this thread starts executing after `tryConnectToAllClusterMembers` . I will fix that.